File: src/test/java/com/koddy/server/common/config/RedisCleanerEachCallbackExtension.java
package com.koddy.server.common.config;

import com.koddy.server.common.utils.RedisCleaner;
import org.junit.jupiter.api.extension.AfterEachCallback;
import org.junit.jupiter.api.extension.ExtensionContext;
import org.springframework.test.context.junit.jupiter.SpringExtension;

public class RedisCleanerEachCallbackExtension implements AfterEachCallback {
    @Override
    public void afterEach(final ExtensionContext context) {
        final RedisCleaner databaseCleaner = SpringExtension.getApplicationContext(context).getBean(RedisCleaner.class);
        databaseCleaner.cleanUpRedis();
    }
}

File: src/test/java/com/koddy/server/common/containers/callback/RedisCleanerEachCallbackExtension.kt
package com.koddy.server.common.containers.callback

import com.koddy.server.common.utils.RedisCleaner
import org.junit.jupiter.api.extension.AfterEachCallback
import org.junit.jupiter.api.extension.ExtensionContext
import org.springframework.test.context.junit.jupiter.SpringExtension

class RedisCleanerEachCallbackExtension : AfterEachCallback {
    override fun afterEach(context: ExtensionContext) {
        SpringExtension.getApplicationContext(context)
            .getBean(RedisCleaner::class.java)
            .cleanUpRedis()
    }
}
